nexus/src/app/mod.rs

```diff
@@ -178,13 +178,22 @@ impl Nexus {
         {
             (dpd_address.ip().to_string(), dpd_address.port())
         } else {
-            let addr = resolver
-                .lock()
-                .await
-                .lookup_socket_v6(ServiceName::Dendrite)
-                .await
-                .map_err(|e| format!("Cannot access Dendrite address: {e}"))?;
-            (addr.ip().to_string(), addr.port())
+            loop {
+                match resolver
+                    .lock()
+                    .await
+                    .lookup_socket_v6(ServiceName::Dendrite)
+                    .await
+                    .map_err(|e| format!("Cannot access Dendrite address: {e}"))
+                {
+                    Ok(addr) => break (addr.ip().to_string(), addr.port()),
+                    Err(e) => {
+                        warn!(log, "Failed to access Dendrite address: {e}");
+                        tokio::time::sleep(std::time::Duration::from_secs(1))
+                            .await;
+                    }
```

Comment:
This seems like something we probably want to revisit.  (Nexus can't come up _at all_ without Dendrite?  It seems like it'd be better to just construct a lazily-resolving DPD client and let the rest of Nexus come up.)

But this PR is a clear improvement from what's there now.